source/Nevermore/TableColumnsCache.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nevermore
+{
+    public class TableColumnsCache : ITableColumnsCache
```

Comment:
This is currently covering two concerns: resolving columns and caching them. Can we split these out and manage them separately?